code/integration-tests/runtime-tests/test/tests/assets/rpcAssetsTests.ts

```diff
@@ -72,6 +80,83 @@ describe("[SHORT] rpc.assets Tests", function () {
     const KSM = result.find(e => hex_to_ascii(e.name.toString()) === "KSM")!;
     expect(PICA.id.toNumber()).to.equal(1);
     expect(KSM.id.toNumber()).to.equal(4);
+    expect(PICA.decimals.toNumber()).to.equal(12);
+
+    // Update KSM
+    let newKsmInfo = {
+      decimals: 2,
+      foreignId: { parents: "2", interior: "Here" },
+      ratio: rational(2, 10),
+      existentialDeposit: 222_000
+    };
+    await sendAndWaitForSuccess(
+      api,
+      sudoKey,
+      api.events.sudo.Sudid.is,
+      api.tx.sudo.sudo(
+        api.tx.assetsRegistry.updateAsset(4, newKsmInfo.foreignId, newKsmInfo.ratio, newKsmInfo.decimals)
+      )
+    );
+
+    let resultAfterUpdate = await RpcAssetsTests.rpcListAssetsTest(api);
+    expect(resultAfterUpdate.length).to.eq(result.length);
+    let KSMAfterUpdate = resultAfterUpdate.find(e => hex_to_ascii(e.name.toString()) === "KSM")!;
+    expect(KSMAfterUpdate).to.not.be.undefined;
+    expect(KSMAfterUpdate.decimals.toNumber()).to.equal(newKsmInfo.decimals);
+    expect(KSMAfterUpdate.foreignId.toHuman()).to.deep.equal(newKsmInfo.foreignId);
+    expect(KSMAfterUpdate.ratio.toJSON()).to.deep.equal(newKsmInfo.ratio);
+
+    // Update KSM
+    newKsmInfo = {
+      decimals: 3,
+      foreignId: { parents: "3", interior: "Here" },
+      ratio: rational(3, 10),
+      existentialDeposit: 333_000
+    };
+    await sendAndWaitForSuccess(
+      api,
+      sudoKey,
+      api.events.sudo.Sudid.is,
+      api.tx.sudo.sudo(
+        api.tx.assetsRegistry.updateAsset(4, newKsmInfo.foreignId, newKsmInfo.ratio, newKsmInfo.decimals)
+      )
+    );
+
+    resultAfterUpdate = await RpcAssetsTests.rpcListAssetsTest(api);
+    KSMAfterUpdate = resultAfterUpdate.find(e => hex_to_ascii(e.name.toString()) === "KSM")!;
+    expect(KSMAfterUpdate).to.not.be.undefined;
+    expect(KSMAfterUpdate.decimals.toNumber()).to.equal(newKsmInfo.decimals);
+    expect(KSMAfterUpdate.foreignId.toHuman()).to.deep.equal(newKsmInfo.foreignId);
+    expect(KSMAfterUpdate.ratio.toJSON()).to.deep.equal(newKsmInfo.ratio);
+
+    // Register new asset
+    const newAssetInfo = {
+      decimals: 4,
+      foreignId: { parents: "4", interior: { X1: { Parachain: "4" } } },
```

Comment:
Nice test. So there is no such Origin defined (4 parents up). So you will not be able to send such asset. Better use 0 or 1. Then can in theory send something later. 

In future XCM will have absolute paths. But not now.

Reply:
Perfect, I'll change `parents` to `1`. What should happen on the pallet side if someone registers an invalid `foreignId` such as this one?

Reply:
Will merge trap.md explaining some soon with docs (right now in PR). Likely you or Dom may work on that some day.